internal/pkg/service/common/configpatch/dump.go

```diff
@@ -30,126 +25,19 @@ type DumpKV struct {
 // Only keys found in both, configuration and patch structure, are processed.
 // The structure is flattened, keys are joined with a dot "." separator.
 // Each key-value pair contains information whether the value was overwritten from the patch or not.
-func DumpKVs(configStruct, patchStruct any, opts ...Option) ([]DumpKV, error) {
-	cfg := newConfig(opts)
-
```

Comment:
I need a big part of the `DumpKV` function in the `Apply` function, so I created new `visitConfigAndPatch` function.